### ISSUE TITLE
asyncForEach typed incorrectly when given list of promises

### ISSUE THREAD

**Issue by IlyaSemenov** [nicolas-van/modern-async#19]
**Describe the bug**
`asyncForEach` (and probably other similar functions) is typed incorrectly when it's given a list of promises. Namely, the `iteratee` callback value is typed as if it were a promise, while in fact it's the resolved value.

**To Reproduce**
```ts
import { asyncForEach } from "modern-async"

const input = Array.from({ length: 10 }).map((_, i) => i + 1) // 1...10

await asyncForEach(
  input.map(id =>
    Promise.resolve().then(async () => id),
  ),
  async (value) => {
    // value here is typed as Promise<number> while in fact it's the actual number
    console.log(value)
  },
)
```

**Expected behavior**
`value` in the callback above should be typed as `number`, reflecting the actual runtime type of value.

**Environment:**
 - OS: macOS
 - Node v22.6.0
 - Typescript 5.5.4
 - modern-async 2.0.0

**Additional context**
I believe here (and in all similar places):
https://github.com/nicolas-van/modern-async/blob/b1b83ae89cae49a100bf58fe9b4aaa1d2efc45e9/modern-async.d.ts#L97

`Iterable<V> | AsyncIterable<V>` should be replaced with `Iterable<V | PromiseLike<V>> | AsyncIterable<V>`

**Comment by IlyaSemenov** [comment]
Thanks! Any idea when this can land in npm?

**Comment by nicolas-van** [comment]
I'll say by the end of the week. I'm still considering to develop some of the opened feature requests to integrate them in version with multiple changes.

**Comment by nicolas-van** [comment]
This fix was published along version 2.0.1 (sorry for the delay).
